feat: OPTIC-1733: Standardize Dropdown Components Using LSE selector

yyassi-heartex opened the pull request:
New unified Select component
![image](https://github.com/user-attachments/assets/38337313-b541-48ed-a70e-7f757cd29e5e)
This pull request primarily focuses on replacing existing `Select` components with the `Select` component from the `@humansignal/ui` library across multiple files. Additionally, it includes some refactoring and enhancements to improve code readability and functionality. Below are the most important changes:

### UI Component Updates:
* Replaced the existing `Select` components with `Select` from `@humansignal/ui` in various files, including `Select.jsx`, `Pagination.tsx`, `Config.jsx`, `CreateProject.jsx`, `GeneralSettings.jsx`, `Forms.jsx`, `StorageForm.jsx`, `SampleDatasetSelect.tsx`, `ModelVersionSelector.jsx`, and `Select.jsx` in the `datamanager` directory. [[1]](diffhunk://#diff-1f97a98feb63442f8e8630535f276501c71e0d1e00e5ecf0a7c474ac1b3afb3eR5) [[2]](diffhunk://#diff-5e0e6b1bf1d95e422e09552ef7bf18836be7c76fe6a54718f65d7395e5325e68L14-R14) [[3]](diffhunk://#diff-b8cade5f083b7ee986ecfffd94abcf8500ea563ec194ccf3ca3d94f2edda8a89L25-R25) [[4]](diffhunk://#diff-37799f8bcc3b80b87f49ceb8ec48563ba527a6597bb0a6cf9d4e5fd6550d836cL1-R1) [[5]](diffhunk://#diff-a7fd651c076ef3072e41d654fa2a75eea9e4e91dc065f7327f3088b59adaa5deL1-R4) [[6]](diffhunk://#diff-ea6cf290f15b98dfcd6f32b3d3fe671cc7d571c09d354b09eb8171e0baa77b3fL9-R9) [[7]](diffhunk://#diff-df61b153718e709aa2440e7ef4fb7bb18b5975152d535432a5572556dafdb50bL44-R45) [[8]](diffhunk://#diff-a5b6a20921ce7c4eaa4b7db73462effa2e02d835b4098c40cb29649fee3754ecL1-R1) Fbb5e0ebL1, [[9]](diffhunk://#diff-154cb11b2d873404a842cff0d25f8b83dff10c2d057ec1c2d463069d004db558R5)

### Functional Enhancements:
* Updated event handlers for `Select` components to use the new `onChange` signature provided by `@humansignal/ui`, replacing the previous event-based handlers. [[1]](diffhunk://#diff-1f97a98feb63442f8e8630535f276501c71e0d1e00e5ecf0a7c474ac1b3afb3eL49-R66) [[2]](diffhunk://#diff-5e0e6b1bf1d95e422e09552ef7bf18836be7c76fe6a54718f65d7395e5325e68L266-R267) [[3]](diffhunk://#diff-b8cade5f083b7ee986ecfffd94abcf8500ea563ec194ccf3ca3d94f2edda8a89L157-R174) [[4]](diffhunk://#diff-33d58198c5fa7595649983ca5e0eb3f454983be6b1134040b991dcf21f5184e1L85-R85) [[5]](diffhunk://#diff-ea6cf290f15b98dfcd6f32b3d3fe671cc7d571c09d354b09eb8171e0baa77b3fL41-R43) [[6]](diffhunk://#diff-df61b153718e709aa2440e7ef4fb7bb18b5975152d535432a5572556dafdb50bL44-R45)

### Code Refactoring:
* Added `useMemo` hooks to optimize the computation of options for `Select` components, ensuring better performance and readability. [[1]](diffhunk://#diff-1f97a98feb63442f8e8630535f276501c71e0d1e00e5ecf0a7c474ac1b3afb3eL27-R44) [[2]](diffhunk://#diff-b8cade5f083b7ee986ecfffd94abcf8500ea563ec194ccf3ca3d94f2edda8a89R277-R304) [[3]](diffhunk://#diff-a5b6a20921ce7c4eaa4b7db73462effa2e02d835b4098c40cb29649fee3754ecL30-R45)

### CSS and Layout Adjustments:
* Modified CSS styles to support the new `Select` component layout, including changes in `uikit.css` to use `display: flex` instead of `display: block`.

### Miscellaneous:
* Minor adjustments to imports and code structure to maintain consistency and improve code clarity. [[1]](diffhunk://#diff-b8cade5f083b7ee986ecfffd94abcf8500ea563ec194ccf3ca3d94f2edda8a89L3-R10) [[2]](diffhunk://#diff-69c27f83b5c2e14a1e4fb411596068407ea0fbb9cbb20adfefb11deb122789e4L12-R13) [[3]](diffhunk://#diff-37799f8bcc3b80b87f49ceb8ec48563ba527a6597bb0a6cf9d4e5fd6550d836cL15-R15) [[4]](diffhunk://#diff-154cb11b2d873404a842cff0d25f8b83dff10c2d057ec1c2d463069d004db558L23-L24)

These changes collectively enhance the consistency and maintainability of the codebase by standardizing the use of the `Select` component from `@humansignal/ui` and optimizing related code.

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements

bmartel committed:
fixing AnnotationsCarousel tests

yyassi-heartex committed:
cleaning up a situation where its possible for user to be null when we try to spread its values

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements

yyassi-heartex committed:
chore: OPTIC-1778: Unify and prettify icons

yyassi-heartex committed:
slight tweaks

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1748/ui-improvements'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697081739

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697081739

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13697520796

yyassi-heartex committed:
test and lint cleanup

yyassi-heartex committed:
Merge branch 'fb-optic-1748/ui-improvements' of github.com:heartexlabs/label-studio into fb-optic-1748/ui-improvements

yyassi-heartex committed:
fixing an issue that makes it impossible for a user to update the fields

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements

yyassi-heartex committed:
Revert "Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements"

This reverts commit 291f6e37083861000c5d3bbb1ff2e73e7119a2ce, reversing
changes made to 1c53cd9462bceb87ee0d9a499be0ec0b1410dce1.

yyassi-heartex committed:
Reapply "Merge remote-tracking branch 'origin/develop' into fb-optic-1748/ui-improvements"

This reverts commit 233340c6576e1af660ec6386eb45c2ac65e6d144.

yyassi-heartex committed:
fixing merge

yyassi-heartex committed:
cleaning up imports

yyassi-heartex committed:
Merge branch 'fb-optic-1748/ui-improvements' into fb-optic-1778/icons

yyassi-heartex committed:
addressing a different in icon usage

yyassi-heartex committed:
icon cleanup

yyassi-heartex committed:
import upload uptton sizing

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13716495646

yyassi-heartex committed:
Update web/libs/core/src/pages/AccountSettings/sections/PersonalInfo.tsx

yyassi-heartex committed:
Update web/libs/ui/src/lib/Userpic/Userpic.tsx

yyassi-heartex committed:
Update web/libs/ui/src/lib/Userpic/Userpic.tsx

yyassi-heartex committed:
unfiying icons

yyassi-heartex committed:
fixing icon

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1748/ui-improvements

yyassi-heartex committed:
cleaned out unused files + fixed tooltip breakage due to quick toggle + added intent delay

yyassi-heartex committed:
lint cleanup

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13763179195

yyassi-heartex committed:
we dont need this icon here

yyassi-heartex committed:
Merge branch 'fb-optic-1748/ui-improvements' into fb-optic-1778/icons

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1778/icons

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1778/icons

yyassi-heartex committed:
icon updates

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
icon cleanup

yyassi-heartex committed:
react-icons no longer used + cleanup

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1778/icons

yyassi-heartex committed:
fixing icon import

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1778/icons

yyassi-heartex committed:
fixing e2e + integration tests

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-1778/icons'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13796443131

yyassi-heartex committed:
slight tweaks

yyassi-heartex committed:
Merge branch 'fb-optic-1778/icons' of github.com:heartexlabs/label-studio into fb-optic-1778/icons

yyassi-heartex committed:
fixing an import

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
unifiying LSE icons into ui library

yyassi-heartex committed:
adjusting Compare all annotations icon size

yyassi-heartex committed:
bigger button looks better

bmartel committed:
fix top bar icon sizing to match design intent

bmartel committed:
comment form icon styles

bmartel committed:
don't increase weight of icons by forcing a stroke to paths of button icons

bmartel committed:
fix eye icon sizing in relations and regions panel

bmartel committed:
linting

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1778/icons

bmartel committed:
fix slider icon import

bmartel committed:
add new icons to replace MD Icons in app

bmartel committed:
added sync disabled icon

yyassi-heartex committed:
updating bell.svg

yyassi-heartex committed:
Merge branch 'fb-optic-1778/icons' of github.com:heartexlabs/label-studio into fb-optic-1778/icons

bmartel committed:
remove react-icons

bmartel committed:
Merge branch 'fb-optic-1778/icons' of github.com:HumanSignal/label-studio into fb-optic-1778/icons

yyassi-heartex committed:
switching to unified select

yyassi-heartex committed:
feat: OPTIC-1733: Standardize Dropdown Components Using LSE selector

yyassi-heartex committed:
switching to use ui select

yyassi-heartex committed:
standardized datamanger filer select

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1778/icons'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13920015044

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13920015044

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
now using new file upload.svg

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1778/icons

yyassi-heartex committed:
resolving merge issue + restoring refresh icon

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
fixing disabled flag

yyassi-heartex committed:
overhauling select to movetowards combobox mehavior

yyassi-heartex committed:
Merge branch 'fb-optic-1778/icons' into fb-optic-1733/select

yyassi-heartex committed:
prepping for cmdk

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1778/icons'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13943734464

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/13943734464

yyassi-heartex committed:
updating modal close icon size

yyassi-heartex committed:
Merge branch 'fb-optic-1778/icons' into fb-optic-1733/select

yyassi-heartex committed:
LSO now fully Select unified

yyassi-heartex committed:
cleaning up unused files

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
fixing a DateTime select issue

yyassi-heartex committed:
testid being added to select

yyassi-heartex committed:
fixing selection

yyassi-heartex committed:
cleaning up dependencies

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
stabilizing merge

yyassi-heartex committed:
we now have working DM filters with search

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
select is functional with a way to deal with inline selects

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
fixed select behavior

yyassi-heartex committed:
making sure we have a ref pointing to an element that populates a parent form

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
slight cleanup

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
select cleanup + label support

yyassi-heartex committed:
slight visual  tweaks for labeled selects, adding test ids and visual updates for select

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
supporting testid prop

yyassi-heartex committed:
tweaking story and lint cleanup

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
modifying visuals and behavior

yyassi-heartex committed:
select is in a stable spot

yyassi-heartex committed:
filter rendering

yyassi-heartex committed:
fixed an issue boolean values

yyassi-heartex committed:
further tweaks

yyassi-heartex committed:
multi select parent can now toggle children on click

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
Select trigger visuals

yyassi-heartex committed:
more visual tweaks

yyassi-heartex committed:
test id support

yyassi-heartex committed:
fixing e2e test

yyassi-heartex committed:
fixing unit test

yyassi-heartex committed:
fixing unit test

yyassi-heartex committed:
slight tweak to trigger props and select className

yyassi-heartex committed:
fixing unit tests

yyassi-heartex committed:
fixing tests

yyassi-heartex committed:
visual fixes

yyassi-heartex committed:
updating e2e

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
tweaking e2e test

yyassi-heartex committed:
updating option visuals + e2e test fix

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
changing how we test for select option

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14172131712)
> Successfully merged:  1 file changed, 38 insertions(+), 8 deletions(-)

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-1733/select'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14172131712

yyassi-heartex committed:
moving along

yyassi-heartex committed:
Merge branch 'fb-optic-1733/select' of github.com:heartexlabs/label-studio into fb-optic-1733/select

yyassi-heartex committed:
this should result in passing e2e

nass600 commented on web/apps/labelstudio/src/pages/Settings/AnnotationSettings/ModelVersionSelector.jsx:
```diff
@@ -82,7 +82,7 @@ export const ModelVersionSelector = ({
             disabled={!versions.length && !models.length}
             value={version}
-            onChange={(e) => setVersion(e.target.value)}
+            onChange={(val) => setVersion(val)}
```
```suggestion
            onChange={setVersion}
```

nass600 commented on web/apps/labelstudio/src/pages/Settings/MachineLearningSettings/Forms.jsx:
```diff
@@ -38,8 +38,9 @@ const CustomBackendForm = ({ action, backend, project, onSubmit }) => {
-          onChange={(e) => {
-            setAuthMethod(e.target.value);
+          value={selectedAuthMethod}
+          onChange={(val) => {
```
```suggestion
          onChange={setAuthMethod}
```

nass600 commented on web/apps/labelstudio/src/pages/Settings/StorageSettings/StorageForm.jsx:
```diff
@@ -41,8 +41,8 @@ export const StorageForm = forwardRef(({ onSubmit, target, project, rootClass, s
         value: storage?.type ?? type,
-        onChange: (e) => {
-          setType(e.target.value);
+        onChange: (value) => {
```
```suggestion
        onChange: setType,
```

nass600 commented on web/libs/datamanager/src/components/CellViews/Annotators/Annotators.jsx:
```diff
@@ -80,6 +80,7 @@ const UsersInjector = inject(({ store }) => {
 Annotators.FilterItem = UsersInjector(({ users, item }) => {
   const user = users.find((u) => u.id === item);
 
+  console.log("user", user);
```
```suggestion
```

yyassi-heartex committed:
Update web/apps/labelstudio/src/pages/Settings/AnnotationSettings/ModelVersionSelector.jsx

Co-authored-by: Ignacio Velazquez <ignacio@humansignal.com>

yyassi-heartex committed:
Update web/apps/labelstudio/src/pages/Settings/MachineLearningSettings/Forms.jsx

Co-authored-by: Ignacio Velazquez <ignacio@humansignal.com>

yyassi-heartex committed:
Update web/apps/labelstudio/src/pages/Settings/StorageSettings/StorageForm.jsx

Co-authored-by: Ignacio Velazquez <ignacio@humansignal.com>

yyassi-heartex committed:
Update web/libs/datamanager/src/components/CellViews/Annotators/Annotators.jsx

Co-authored-by: Ignacio Velazquez <ignacio@humansignal.com>

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
fix due to replacement

yyassi-heartex committed:
clearing out tests that are irrelevant because it is on old region item stuff instead of outliner

yyassi-heartex committed:
adding data-value attribute to select trigger

yyassi-heartex committed:
cleaning up usage and standardizing visuals and behavior

yyassi-heartex committed:
the DM select component is only used in DM Pagination which is not used

yyassi-heartex committed:
cleaning up

yyassi-heartex committed:
updating paragraphs test

yyassi-heartex committed:
fixing the selector

yyassi-heartex committed:
updating paragraphs test

AndrejOros commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14189757624)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1733/select'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14189757624

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14189757624

yyassi-heartex committed:
applying tweak for visuals for initial feedback + making testing easier

yyassi-heartex committed:
fixing e2e test

yyassi-heartex committed:
fixing more tests

luarmr committed:
experiment with claude

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
cleanup

yyassi-heartex committed:
cleanup

yyassi-heartex committed:
cleaning up select input

yyassi-heartex committed:
cleaning up selects

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
now supporting sizes

yyassi-heartex committed:
getting model version selector working in LSO

yyassi-heartex committed:
updating testid

yyassi-heartex committed:
using hidden text input + updating test id for tests

yyassi-heartex committed:
we now dispatch the change event

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
cleaning up placeholder visibility

yyassi-heartex committed:
fixing a select issue where it flows past the edge in some situations

yyassi-heartex committed:
select supports custom filter, searchInput on change and datamanager annotator select will now filter by display name or email or id

yyassi-heartex committed:
updating select visuals

nick-skriabin commented:
one suggestion for the future improvement

when select has search enabled it'd be cool that the select trigger become the input itself rather than having an input inside a dropdown. so it looks more like an autocomplete but will limit what you can select in it to only be the list of items in the select. essentially, it allows free form input like a input field, but will not allow leaving random value. instead, upon selecting one of the options it will fill the input with the value of the option

from what I noticed in the modern UIs, it became a standard for the dropdowns/selects like this

an example of this would be this component https://react-select.com/home

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex commented:
> one suggestion for the future improvement
> 
> when select has search enabled it'd be cool that the select trigger become the input itself rather than having an input inside a dropdown. so it looks more like an autocomplete but will limit what you can select in it to only be the list of items in the select. essentially, it allows free form input like a input field, but will not allow leaving random value. instead, upon selecting one of the options it will fill the input with the value of the option
> 
> from what I noticed in the modern UIs, it became a standard for the dropdowns/selects like this
> 
> an example of this would be this component https://react-select.com/home

I love this idea! I'll set up a follow up ticket to include product, design and implement this

yyassi-heartex committed:
lint cleanup

yyassi-heartex commented on web/apps/labelstudio/src/pages/CreateProject/Import/Import.jsx:
```diff
@@ -9,8 +9,8 @@ import { useAtomValue } from "jotai";
 import { Modal } from "../../../components/Modal/Modal";
-import { useAPI } from "../../../providers/ApiProvider";
 import { cn } from "../../../utils/bem";
+import { useAPI } from "../../../providers/ApiProvider";
```
revert the import line order so its alphabetical by path

yyassi-heartex committed:
applying feedback + lint cleanup

yyassi-heartex committed:
making sure we wait a little before we do the next step to avoid failure in ci

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
updated visuals

yyassi-heartex committed:
select forcing a change event

yyassi-heartex committed:
altering the hidden select

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
fixing filterRow

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
fixing date-time test to match updated select

yyassi-heartex committed:
fixing filter.test

yyassi-heartex committed:
fixing filter.test

yyassi-heartex committed:
lint cleanup

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14346628715)
> Successfully merged: Already up to date.

yyassi-heartex committed:
forcing an update

yyassi-heartex committed:
forcing an update

yyassi-heartex committed:
resizeObservable

yyassi-heartex committed:
filter test fix

yyassi-heartex committed:
we nee to make sure on change fires on change - not only on event dispatch

yyassi-heartex committed:
updating filter test

yyassi-heartex committed:
visual tweaks to select

yyassi-heartex committed:
event handling should be better now

yyassi-heartex committed:
addressing weirdness with datamanger selects

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
is this redundant?

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14379835362)
> Successfully merged:  1 file changed, 57 insertions(+), 3 deletions(-)

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-1733/select'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14379835362

yyassi-heartex committed:
now supporting selected value render

yyassi-heartex committed:
darkmode support

yyassi-heartex committed:
header support

yyassi-heartex committed:
Merge branch 'develop' into fb-optic-1733/select

yyassi-heartex committed:
fixing merge

yyassi-heartex committed:
adding test ids

yyassi-heartex committed:
trigger test id

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
reverting initial changes - trying again

yyassi-heartex committed:
author filter is now multi select as intially intended

yyassi-heartex committed:
paragraphs tests now succesful

yyassi-heartex committed:
making sure first option is selected

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
tweaking display for feature parity

yyassi-heartex committed:
lint cleanup

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14464548326)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1733/select'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14464548326

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14464548326

yyassi-heartex committed:
minor tweak to preselect first option if no placeholder provided

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
updating focus visual

yyassi-heartex committed:
undoing defaulting to first option as it breaks behavior in other spots

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
cleaning up icons

yyassi-heartex committed:
using corrected search icon

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
slight tweak to the popover content so it has a max-width of available space

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
slight visual tweaks based on feedback

yyassi-heartex commented on web/libs/datamanager/src/components/Filters/FilterInput.jsx:
```diff
@@ -11,13 +11,12 @@ export const FilterInput = ({ value, type, onChange, placeholder, schema, style
       placeholder={placeholder}
       onChange={onChangeHandler}
-      style={style}
+      style={{ height: "100%", ...style }}
```
change to use classname h-full instead of inline style 100%

yyassi-heartex commented on web/libs/datamanager/src/components/Filters/FilterLine/FilterLine.scss:
```diff
@@ -10,15 +10,14 @@
     justify-content: center;
 
     .button-dm {
+      height: 100%;
```
set width to 24px to make sure our button is a square

yyassi-heartex replied:
```suggestion
      height: 100%;
      width: 24px;
      padding: 0;
```

yyassi-heartex committed:
Update web/libs/datamanager/src/components/Filters/FilterLine/FilterLine.scss

yyassi-heartex committed:
applying changes based on feedback

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14511227706)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1733/select'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14511227706

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14511227706

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14519405375)
> Successfully merged:  10 files changed, 21 insertions(+), 18 deletions(-)

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-1733/select'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14519405375

yyassi-heartex commented on web/libs/ui/src/lib/select/select.tsx:
```diff
@@ -0,0 +1,416 @@
+
+        let selectedOptions = [];
+        if (multiple) {
+          setValue((prev = []) => {
+            if (isSelected) {
+              return (selectedOptions = [...prev.filter((v) => v !== val)]);
+            }
+            return (selectedOptions = [...prev, val]);
+          });
```
this could cause an issue with the callback handler getting called after passing selectedOptions up to props.onChange. use a valueRef instead of the callback

yyassi-heartex commented on web/libs/ui/src/lib/select/select.tsx:
```diff
@@ -0,0 +1,416 @@
+            )}
+          </button>
+        </PopoverTrigger>
+        <PopoverContent className="z-99999 min-w-full" align="start" data-testid="select-popup">
```
```suggestion
        <PopoverContent align="start" data-testid="select-popup">
```

yyassi-heartex commented on web/libs/ui/src/lib/select/select.tsx:
```diff
@@ -0,0 +1,416 @@
+                              onSelect={() => {
+                                children.forEach((child: SelectOption<T>) => {
+                                  const childVal = child?.value ?? child;
+                                  isOptionSelected ? _onChange(childVal, true) : _onChange(childVal, false);
```
```suggestion
                                  _onChange(childVal, isOptionSelected);
```

yyassi-heartex committed:
tweaking searching group container for Select + removing radix-ui select

borisheartex commented:
/fm sync
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14532437125)

robot-ci-heartex committed:
Merge branch 'develop' into 'fb-optic-1733/select'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14532437125

robot-ci-heartex committed:
Sync Follow Merge dependencies

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14532437125

yyassi-heartex committed:
Update web/libs/ui/src/lib/select/select.tsx

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
fixing issue detected in QA

yyassi-heartex committed:
Update web/libs/ui/src/lib/select/select.tsx

yyassi-heartex committed:
lint cleanup

yyassi-heartex committed:
Merge branch 'fb-optic-1733/select' of github.com:heartexlabs/label-studio into fb-optic-1733/select

yyassi-heartex commented:
/git merge develop
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14598338263)
> Successfully merged:  1 file changed, 10 insertions(+), 8 deletions(-)

yyassi-heartex committed:
Merge branch 'develop' into 'fb-optic-1733/select'

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/14598338263

yyassi-heartex committed:
making sure we update the story

yyassi-heartex commented:
/fm sync

yyassi-heartex commented:
/git merge
> [Workflow run](https://github.com/HumanSignal/label-studio/actions/runs/14599855607)
> Successfully merged: Already up to date.

bmartel committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select

yyassi-heartex committed:
taking leading-none out

yyassi-heartex committed:
Merge remote-tracking branch 'origin/develop' into fb-optic-1733/select
